File: fast_sync_dotnet_service/fast-sync-core/abstraction/metadata/ISyncableObject.cs
namespace fast_sync_core.abstraction.data
{
    public interface ISyncableObject<T> : IWithId
        where T : ISyncMetadata
    {
        T Metadata { get; set; }
        bool Deleted { get; set; }
        int GetVersion();
        int SetVersion(int version);
    }
}

File: fast_sync_typescript_service_sdk/packages/fast-sync-core/src/core/abstraction/metadata/ISyncable_object.ts
import { ISyncMetadata } from "./ISync_metadata";
import { IWithId } from "./Iwith_id";

export interface ISyncableObject extends IWithId  {
    id:string;
    metadata: ISyncMetadata;
    deleted: boolean;
    getVersion():number;
    setVersion(version: number):number;
}
  
